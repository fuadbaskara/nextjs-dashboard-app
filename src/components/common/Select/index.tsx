import React from 'react';

export type SelectProps = {
  title: string;
  name: string;
  onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
  options: SelectOption[];
  value?: string;
  defaultValue?: string;
  placeholder?: string;
};

export type SelectOption = {
  label: string;
  value: string;
};

function Select({ title, name, onChange, options, value, defaultValue, placeholder }: SelectProps) {
  return (
    <div>
      <label htmlFor={name} className="block text-sm font-medium text-gray-700">
        {title}
      </label>
      <select
        id={name}
        name={name}
        className="mt-1 block w-full min-w-[200px] rounded-md border-gray-300 py-2 pl-3 pr-10 text-base shadow-sm focus:border-indigo-500 focus:outline-none focus:ring-indigo-500 sm:text-sm"
        value={value || defaultValue}
        placeholder={placeholder}
        onChange={onChange}
      >
        {(options || []).map((option: SelectOption) => (
          <option key={option.value} value={option.value}>
            {option.label}
          </option>
        ))}
      </select>
    </div>
  );
}

Select.defaultProps = {
  value: '',
  defaultValue: '',
  placeholder: '',
};

export default Select;
