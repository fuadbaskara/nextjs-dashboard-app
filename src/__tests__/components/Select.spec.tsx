/* eslint-disable testing-library/no-node-access */
import React from 'react';
import { render } from '../test-utils';
import Select from '@/components/common/Select';

const options = [
  {
    label: 'All',
    value: '',
  },
  {
    label: 'Male',
    value: 'male',
  },
  {
    label: 'Female',
    value: 'female',
  },
];

const renderSelect = () =>
  render(
    <Select
      title="Gender"
      name="gender"
      value="male"
      options={options}
      defaultValue="all"
      onChange={() => jest.fn()}
    />,
  );

describe('Select Component', () => {
  test('renders correctly with simple props:', () => {
    const { asFragment } = renderSelect();
    const select = document.querySelector('select[name="gender"]');
    expect(select).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
