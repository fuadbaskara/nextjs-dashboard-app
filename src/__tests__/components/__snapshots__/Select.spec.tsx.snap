// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component renders correctly with simple props: 1`] = `
<DocumentFragment>
  <div>
    <label
      class="block text-sm font-medium text-gray-700"
      for="gender"
    >
      Gender
    </label>
    <select
      class="mt-1 block w-full min-w-[200px] rounded-md border-gray-300 bg-white py-2 pl-3 pr-10 text-base shadow-sm outline-none hover:cursor-pointer sm:text-sm"
      id="gender"
      name="gender"
      placeholder=""
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="male"
      >
        Male
      </option>
      <option
        value="female"
      >
        Female
      </option>
    </select>
  </div>
</DocumentFragment>
`;
